_posts/2022-12-14-#26631.md

```diff
@@ -10,17 +10,117 @@ status: upcoming
+- There are several kinds of outputs (for example, P2PK, P2PKH), and each has
+  a characteristic size, both of the output itself and for the input that will
+  later spend it. Enforcing the dust limit requires the code to
+  [calculate](https://github.com/bitcoin/bitcoin/blob/1ea02791f3d81c7716d9ea455971203f74d7a107/src/policy/policy.cpp#L26)
+  the sizes of the various kinds of inputs and outputs.
+
+- The concept of dust was first introduced in
+  [PR 2577](https://github.com/bitcoin/bitcoin/pull/2577) in 2013.
+  This [commit](https://github.com/bitcoin/bitcoin/pull/9380/commits/eb30d1a5b215c6dd3763d7f7948f2dd8cb61f6bf)
+  from [PR 9380](https://github.com/bitcoin/bitcoin/pull/9380)
+  introduced the `-dustrelayfee` option.
+  Previous to that PR, the dust feerate was whatever `-minrelaytxfee` was set to.
+
 ## Questions
 
-1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
+1. Did you review the PR?
+   [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
+
+1. How does the test work? What is its approach?
+
+1. Why is the concept of _dust_ useful? What problems might occur if it didn't exist?
+
+1. A transaction with an output considered "dust" is classified as a _valid_,
+   but _non-standard_, transaction.
+   What is the difference between valid transaction and a non-standard transaction?
+   Would it be better if transactions that don't meet the dust treshhold
+   were considered invalid?
+
+1. Why does the dust feerate limit apply to each output individually, rather than
+   to all of a transaction's outputs collectively?
+   Can you think of an anomolous case in which this policy conflicts with
+   being miner-incentive compatatible?
+
+1. Why is this feerate a configuration option, which makes it fairly static
+   (most node operators probably just accept the default), rather than having
+   it dynamically track the prevailing network feerate?
+
+1. Why is `-dustrelayfee` a _hidden_ (or _debug_) option?
+
+1. Since `-dustrelayfee` is a per-node configuration option, what happens if various
+   nodes on the network set different values?
+
+1. Can you see a future scenario where we'd want to change the default value of `-dustrelayfee`?
+   Would it more likely be increased or decreased? What does this depend on and which other
+   configuration options would then also very likely be adapted?
+
+1. What does the largest possible output script that adheres to standardness rules look like?
+   Is it currently implemented in the functional test?
+
+1. Which of the output scripts need to be inferred from an actual public key (derived from ECKey
+   in the test)? Could some of them also be created with only random data?
+   (Idea: learn which output script types are checked for full EC-validity.
+   Particulary interesting here is the difference between P2PK and P2TR.)
```

Comment:
```suggestion
```
I just verified and unfortunately had this wrong, neither the P2PK nor the P2TR output types need the contained pubkeys to be fully valid (i.e. on the curve) to be treated as standard. Which in turn means that literally all output scripts could be created with random (hashes or pubkey coordinates) data, and we wouldn't even need to involve the ECKey class. Could keep the part before though as trick questions if you want? (The answer would be: none of them, all of them.)